pil-analyzer/src/statement_processor.rs

```diff
@@ -437,20 +449,34 @@ where
     fn resolve_type_name(&self, mut n: TypeName<parsed::Expression<T>>) -> Result<Type, EvalError> {
         // Replace all expressions by number literals.
         for e in n.expressions_mut() {
-            let v = self.evaluate_expression(e.clone())?;
-            *e = parsed::Expression::Number(v);
+            let v = self.evaluate_expression_to_int(e.clone())?;
+            let v_u64 = v.to_u64().ok_or(EvalError::TypeError(format!(
```

Comment:
why u64?

Reply:
This is a number literal inside a type scheme, so it must be an array length. Will add a comment.